config/config_test.go

```diff
@@ -304,6 +304,39 @@ func TestConfig_Finalize(t *testing.T) {
 }
 
 func TestConfig_Validate(t *testing.T) {
+	valid := longConfig.Copy()
+
+	// 2 tasks using same provider w/ auto_commit enabled (should err)
+	autoCommit := valid.Copy()
+	autoCommit.TerraformProviders = &TerraformProviderConfigs{
+		&TerraformProviderConfig{"X": map[string]interface{}{"auto_commit": true}}}
+	ts := *autoCommit.Tasks
+	t2 := ts[0].Copy()
+	*t2.Name = "task2"
+	ts = append(ts, t2)
+	autoCommit.Tasks = &ts
+
+	// task configured with provider that doesn't exist
+	noProvider := *valid.Copy()
+	noProvider.TerraformProviders = &TerraformProviderConfigs{}
+
+	// task configured with provider that doesn't exist
```

Comment:
this comment doesn't quite match my understanding of this config. maybe a copy-pasta from line 319?

Reply:
Total c-n-p error. :facepalm: